src/api/controllers/PostController.ts

```diff
@@ -56,18 +55,33 @@ export class PostController {
     return { posts: await this.postService.filterPosts(filterPostsRequest) };
   }
 
-  @Get('save/userId/:userId/postId/:postId/') 
-  async savePost(@Params() params: PostAndUserUuidParam): Promise<GetPostResponse> {
-    return { post: await this.postService.savePost(params) }; 
+  @Get('archive/')
+  async getArchivedPosts(): Promise<GetPostsResponse> {
+    return { posts: await this.postService.getArchivedPosts() };
   }
 
-  @Get('unsave/userId/:userId/postId/:postId/') 
-  async unsavePost(@Params() params: PostAndUserUuidParam): Promise<GetPostResponse> {
-      return { post: await this.postService.unsavePost(params) };
+  @Get('archive/userId/:id/')
+  async getArchivedPostsByUserId(@Params() params: UuidParam): Promise<GetPostsResponse> {
+    return { posts: await this.postService.getArchivedPostsByUserId(params) };
   }
-  
-  @Get('isSaved/userId/:userId/postId/:postId/') 
-  async isSavedPost(@Params() params: PostAndUserUuidParam): Promise<IsSavedPostResponse> {
-    return { isSaved: await this.postService.isSavedPost(params) }; 
+
+  @Get('archive/postId/:id/')
+  async archivePost(@CurrentUser() user: UserModel, @Params() params: UuidParam): Promise<GetPostResponse> {
+    return { post: await this.postService.archivePost(user, params) };
+  }
+
+  @Get('save/postId/:id/')
```

Comment:
this doesn't seem related to this PR but because it's here I'll comment that I don't see the need for 3 different endpoints to save, unsave, and check if a post is saved. One endpoint that takes in a boolean to save/unsave and a getter to check if a post is saved would've been better + CRUD-like